tests/Cms/Models/ModelWithContentTest.php

```diff
@@ -38,8 +38,81 @@ public function root(): ?string
     }
 }
 
+class BlueprintsModelWithContent extends ExtendedModelWithContent
+{
+    protected $testModel;
+
+    public function __construct(Model $model)
+    {
+        $this->testModel = $model;
+    }
+
+    public function blueprint()
+    {
+        return new Blueprint([
+            'model'  => $this->testModel,
+            'name'   => 'model',
+            'title'  => 'Model',
+            'columns' => [
+                [
+                    'sections' => [
+                        'pages' => [
+                            'name' => 'pages',
+                            'type' => 'pages',
+                            'templates' => [
+                                'foo',
+                                'bar',
+                            ]
+                        ],
+                        'menu' => [
+                            'name' => 'menu',
+                            'type' => 'pages',
+                            'templates' => [
+                                'home',
+                                'default',
+                            ]
+                        ]
+                    ]
+                ]
+            ]
+        ]);
+    }
+}
+
 class ModelWithContentTest extends TestCase
 {
+    public function modelsProvider(): array
+    {
+        $app = new App([
+            'site' => [
+                'children' => [
+                    [
+                        'slug'  => 'foo',
+                        'files' => [
+                            ['filename' => 'a.jpg'],
+                            ['filename' => 'b.jpg']
+                        ]
+                    ]
+                ],
+                'files' => [
+                    ['filename' => 'c.jpg']
+                ]
+            ],
+            'users' => [
+                [
+                    'email' => 'test@getkirby.com'
+                ]
+            ]
+        ]);
+
+        return [
+            [$app->site()],
+            [$app->page('foo')],
+            // [$app->site()->files()->first()], // not testable yet but should be
+            // [$app->user('test@getkirby.com')] // not testable yet but should be
+        ];
```

Comment:
I couldn't write unit test for `File` and `User` model 😞

Reply:
What's the problem with those files?

Reply:
You will see better the error if you remove the comments and run the test.

Reply:
Ugh, I had a look at it, but honestly I have no clue what's going on here.

Reply:
Ha, I solved it. You've setup pages sections for the test and they don't have a parent option. But pages sections in file or user views have to have a parent to be able to fetch pages correctly. This is what caused the error. I will commit a fix in a second.